src/openrct2-ui/windows/StaffFirePrompt.cpp

```diff
@@ -39,71 +43,57 @@ static rct_widget window_staff_fire_widgets[] = {
     WIDGETS_END,
 };
 
-static void WindowStaffFireMouseup(rct_window *w, WidgetIndex widgetIndex);
-static void WindowStaffFirePaint(rct_window *w, rct_drawpixelinfo *dpi);
-
-//0x9A3F7C
-static WindowEventList window_staff_fire_events([](auto& events)
-{
-    events.mouse_up = &WindowStaffFireMouseup;
-    events.paint = &WindowStaffFirePaint;
-});
 // clang-format on
 
-/** Based off of rct2: 0x6C0A77 */
-rct_window* WindowStaffFirePromptOpen(Peep* peep)
+class StaffFirePromptWindow final : public Window
 {
-    rct_window* w;
-
-    // Check if the confirm window already exists.
-    w = window_bring_to_front_by_number(WindowClass::FirePrompt, peep->sprite_index.ToUnderlying());
-    if (w != nullptr)
+public:
+    void Init(rct_windownumber windownumber)
     {
-        return w;
+        number = windownumber;
+        frame_no = 0;
     }
 
-    w = WindowCreateCentred(WW, WH, &window_staff_fire_events, WindowClass::FirePrompt, WF_TRANSPARENT);
-    w->widgets = window_staff_fire_widgets;
-
-    WindowInitScrollWidgets(*w);
-
-    w->number = peep->sprite_index.ToUnderlying();
-
-    return w;
-}
+    void OnOpen() override
+    {
+        widgets = window_staff_fire_widgets;
+        WindowInitScrollWidgets(*this);
+    }
 
-/**
- *
- *  rct2: 0x006C0B40
- */
-static void WindowStaffFireMouseup(rct_window* w, WidgetIndex widgetIndex)
-{
-    switch (widgetIndex)
+    void OnMouseUp(WidgetIndex widgetIndex) override
     {
-        case WIDX_YES:
+        switch (widgetIndex)
         {
-            auto staffFireAction = StaffFireAction(EntityId::FromUnderlying(w->number));
-            GameActions::Execute(&staffFireAction);
-            break;
+            case WIDX_YES:
+            {
+                auto staffFireAction = StaffFireAction(EntityId::FromUnderlying(number));
+                GameActions::Execute(&staffFireAction);
+                break;
+            }
+            case WIDX_CLOSE:
+            case WIDX_CANCEL:
+                Close();
+                break;
         }
-        case WIDX_CANCEL:
-        case WIDX_CLOSE:
-            window_close(*w);
     }
-}
 
-/**
- *
- *  rct2: 0x006C0AF2
- */
-static void WindowStaffFirePaint(rct_window* w, rct_drawpixelinfo* dpi)
-{
-    WindowDrawWidgets(*w, dpi);
+    void OnDraw(rct_drawpixelinfo& dpi) override
+    {
+        DrawWidgets(dpi);
 
-    Peep* peep = GetEntity<Staff>(EntityId::FromUnderlying(w->number));
-    auto ft = Formatter();
-    peep->FormatNameTo(ft);
+        Peep* peep = GetEntity<Staff>(EntityId::FromUnderlying(number));
+        auto ft = Formatter();
+        peep->FormatNameTo(ft);
 
-    ScreenCoordsXY stringCoords(w->windowPos.x + WW / 2, w->windowPos.y + (WH / 2) - 3);
-    DrawTextWrapped(dpi, stringCoords, WW - 4, STR_FIRE_STAFF_ID, ft, { TextAlignment::CENTRE });
+        ScreenCoordsXY textCoords(windowPos.x + WW / 2, windowPos.y + (WH / 2) - 3);
```

Comment:
```suggestion
        ScreenCoordsXY textCoords(windowPos + ScreenCoordsXY{ WW / 2, (WH / 2) - 3) };
```

This can be "simplified" as such

Reply:
"Simplified" in new commit.